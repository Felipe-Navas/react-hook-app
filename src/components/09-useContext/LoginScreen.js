import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  const { user, setUser } = useContext(UserContext);

  return (
    <div>
      <h1>Login</h1>
      <hr />
      <pre>{JSON.stringify(user, null, 3)}</pre>
      <button
        className="btn btn-primary"
        onClick={() => {
          setUser({ id: 123, name: 'Felipe' });
        }}
      >
        Login
      </button>
    </div>
  );
};
