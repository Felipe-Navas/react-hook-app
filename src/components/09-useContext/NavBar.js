import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <div className="container-fluid">
        <Link to="/" className="navbar-brand">
          useContext
        </Link>
        <button
          className="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarNavAltMarkup"
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
          <div className="navbar-nav">
            <NavLink
              to="/"
              className={({ isActive }) =>
                'nav-link ' + (isActive ? 'active' : '')
              }
              aria-current="page"
            >
              Home
            </NavLink>
            <NavLink
              to="/about"
              className={({ isActive }) =>
                'nav-link ' + (isActive ? 'active' : '')
              }
            >
              About
            </NavLink>
            <NavLink
              to="/login"
              className={({ isActive }) =>
                'nav-link ' + (isActive ? 'active' : '')
              }
            >
              Login
            </NavLink>
          </div>
        </div>
      </div>
    </nav>
  );
};
