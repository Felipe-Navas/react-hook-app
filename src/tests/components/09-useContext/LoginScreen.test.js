import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

configure({ adapter: new Adapter() });

describe('Testing the <LoginScreen /> component', () => {
  const user = { name: 'Felipao', email: 'felipenavas.itec@gmail.com' };
  const setUser = jest.fn();
  const wrapper = mount(
    <UserContext.Provider
      value={{
        user,
        setUser,
      }}
    >
      <LoginScreen />
    </UserContext.Provider>
  );
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should run the setUser function with the expected parameter', () => {
    const parameters = { id: 123, name: 'Felipe' };

    wrapper.find('button').simulate('click');

    expect(setUser).toHaveBeenCalledWith(parameters);
    expect(setUser).toHaveBeenCalledTimes(1);
  });
});
