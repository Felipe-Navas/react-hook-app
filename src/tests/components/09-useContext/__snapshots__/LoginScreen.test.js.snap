// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <LoginScreen /> component should render correctly 1`] = `
<div>
  <h1>
    Login
  </h1>
  <hr />
  <pre>
    {
   "name": "Felipao",
   "email": "felipenavas.itec@gmail.com"
}
  </pre>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
